src/frontend/src/statement/copy_table_from.rs

```diff
@@ -66,6 +75,137 @@ impl StatementExecutor {
 
         let entries = lister.list().await.context(error::ListObjectsSnafu)?;
 
+        Ok((object_store, entries))
+    }
+
+    async fn infer_schema(
+        &self,
+        format: &Format,
+        object_store: ObjectStore,
+        path: &str,
+    ) -> Result<SchemaRef> {
+        match format {
+            Format::Csv(format) => Ok(Arc::new(
+                format
+                    .infer_schema(&object_store, path.to_string())
```

Comment:
I think we should change the signature of `infer_schema` to
```rust
async fn infer_schema(&self, store: &ObjectStore, path: &str) -> Result<Schema>
```

The `path.to_string()` is unnecessary.